### ISSUE TITLE
Autoloading Attribute

### ISSUE THREAD

**Issue by GoodPro712** [tModLoader/tModLoader#947]
### Description

An attribute on classes preventing or enabling them to autoload

### What does this proposal attempt to solve or improve?

Attributes will be less code and cleaner code rather than overriding a method and returning.

### Which (other) solutions should be considered?

N/A

### Examples
```cs
//This item will not autoload
[Autoload(false)]
public class CoolItem : ModItem { }
```

### Notes
Could be paired with the #848 pull request

**Comment by Jofairden** [comment]
Not sure if I'm a fan, the virtual method is familiarized and follows the TML paradigm

**Comment by Chicken-Bones** [comment]
Worth discussing, there's room for improvement in autoloading, and having it work well with constructors

**Comment by Jofairden** [comment]
@Chicken-Bones how does this compete with/compare to IAutoLoadable ?

**Comment by Chicken-Bones** [comment]
I need to look into it more, but the current system doesn't work well with Items that want to have custom constructors, because you have to leave a default one just to allow tML to know you _don't_ want it autoloaded

**Comment by Mirsario** [comment]
This has been implemented by Rartrin on the 1.4 branch.
